In [1]:
from sympy import *
import numpy as np
x = symbols('x')

In [2]:
eq=(1+(1/x))**x
expr=limit((eq),x,oo)
expr

E

$$\lim _{n\to \infty }\left(1+{\frac {1}{n}}\right)^{n}$$

In [3]:
expr=float(expr)
expr

2.718281828459045

If you think about this from the perspective of a derivative, what is being implied above is that $e^x$ always changes by exactly the amount that we have.  Recall that the derivative is $$f'(x)=\lim_{dx \to 0}\frac{f(x+dx)-f(x)}{dx}$$<br><br>
$e^x$ is really just $$1+x+\frac{x^2}{2!}+\frac{x^3}{3!}+\frac{x^4}{4!}+ ...$$<brxx>
If you take the derivative of each term you get,
$$e^x=0+1+\frac{2x}{2!}+\frac{3x^2}{3!}+\frac{4x^3}{4!}+...$$<br>
which reduces to,
$$e^x=1+x+\frac{x^2}{2!}+\frac{x^3}{3!}+\frac{x^4}{4!}+...$$<br>
<b>So $e^x$ acts like its own derivative.</b><br><br> 
What is the implication here? Simply this: the base for growth using the function $e$ is the unit quantity earning unit interest for unit time compounded as fast as possible.

The equation shown in line 1 of <code>In [2]:</code> converges on $2.718281828459045...$ which is the base of the natural logarithm, $e$.

Now if you don't have a background in limits or derivitives, the foregoing will not make a lot of sense, but the equation,<br><br> $$1+x+\frac{x^2}{2!}+\frac{x^3}{3!}+\frac{x^4}{4!}+ ...$$<br><br> will.  Simply imagine the value of 1 inserted into the variable x. Let's recall what $2!$ or $3!$ mean.  This is the factorial function and it is defined,<br><br> $${n!=n\times (n-1)\times (n-2)\times (n-3)\times \cdots \times 3\times 2\times 1\,.}$$<br><br>  Taking an example suppose we wanted to calculate $4!$.<br><br>$$4!= 4 \times 3 \times 2 \times 1=24$$

In [4]:
import decimal


def factorial(n):
    factorials = [1]
    for i in range(1, n+1):
        factorials.append(factorials[i-1]*i)
    return factorials


def compute_e(n):
    decimal.getcontext().prec=n+1
    e = 2
    factorials = factorial(2*n+1)
    for i in range(1, n+1):
        counter = 2 * i+2
        denom = factorials[2*i+1]
        e += decimal.Decimal(counter/denom)
    return e


while True:
    n = 101
    if n >= 0 and n <= 1000:
        break

print(str(compute_e(n))[:n+1])

2.7182818284590452011255216367278836969535858146069330362037419521518120894012649347888439864296302599


Here is a plot of $log_e$: 

![Natural log plot](./e.jpg)

Here is a plot of $log_{10}$ for comparison:

![Base 10 log plot](./base10.jpg)

Compare the growth (y-axis) on each.